### ISSUE TITLE
[2.1] Reproducible "internal error in the .NET Runtime" when Tiered JIT is enabled for .NET Core 2.1

### ISSUE THREAD

**Issue by adamsitnik** [dotnet/coreclr#22919]
@kouvel @noahfalk we are currently trying to enable Tiered and Non-Tiered JIT perf runs for all the microbenchmarks that we have in the perf repo.

Our CI runs all the benchmarks for 2.0/2.1/2.2/3.0 and  Full .NET Framework once to make sure that the PRs are not breaking anything.

To my suprise when we have enabled Tiered runs for 2.1 the CI failed. https://github.com/dotnet/performance/pull/336

Full repro:

```cmd
dotnet run -c Release -f netcoreapp2.1
```

```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp2.1</TargetFramework>
    <TieredCompilation>true</TieredCompilation>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="System.Numerics.Vectors" Version="4.5.0" />
  </ItemGroup>
</Project>
```

```cs
using System;
using System.Numerics;

namespace ConsoleApp83
{
    class Program
    {
        static void Main(string[] args)
        {
            var sut = new Perf_Vector3();
            
            GC.KeepAlive(sut.AddOperatorBenchmark());
            
            Console.WriteLine("OK!");
        }
    }

    public static class VectorTests
    {
        public static readonly Vector3 Vector3Delta = new Vector3(1.0f, -1.0f, 1.0f);

        public static readonly Vector3 Vector3Value = new Vector3(-1.0f, 1.0f, -1.0f);
    }
    
    public class Perf_Vector3
    {
        public Vector3 AddOperatorBenchmark() => VectorTests.Vector3Value + VectorTests.Vector3Delta;
    }
}
```

The app just crashes, in Event Log I can find:

```log
Application: tier.exe
CoreCLR Version: 4.6.27019.6
Description: The process was terminated due to an internal error in the .NET Runtime at IP 00007FF90A00B139 (00007FF909E10000) with exit code c0000005.
```

and

```
Faulting application name: tier.exe, version: 1.0.0.0, time stamp: 0x5afb4a19
Faulting module name: coreclr.dll, version: 4.6.27019.6, time stamp: 0x5bc9ebee
Exception code: 0xc0000005
Fault offset: 0x00000000001fb139
Faulting process id: 0x3a68
Faulting application start time: 0x01d4cf67fd13508d
Faulting application path: C:\Projects\tier\bin\Release\netcoreapp2.1\tier.exe
Faulting module path: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\2.1.6\coreclr.dll
Report Id: a66aa205-bd63-4c6f-9612-a032b9e049b3
Faulting package full name: 
Faulting package-relative application ID: 
```

this is not a high priority thing for us, I just want to make sure that you are aware of this bug.

**Comment by adamsitnik** [comment]
I forgot to mention the most important thing: it works fine with .NET Core 3.0 and 2.2

**Comment by kouvel** [comment]
There were a number of issues that were fixed in 3.0 and ported to 2.2. This may be a duplicate of https://github.com/dotnet/coreclr/pull/19237. CC @CarolEidt

**Comment by kouvel** [comment]
If this doesn't need to be fixed in 2.1/2.0 it may be appropriate to disable the vector perf tests when running against those versions.

**Comment by adamsitnik** [comment]
> If this doesn't need to be fixed in 2.1/2.0 it may be appropriate to disable the vector perf tests when running against those versions with tiering enabled.

From our perspective, it does need to be. I have reported this problem so you can investigate if the source of this problem is something important and needs to be fixed in 2.1 or not.

**Comment by noahfalk** [comment]
@adamsitnik did you mean?

> From our perspective, it does **not** need to be.

**Comment by RussKeldorph** [comment]
Closing since there isn't an immediate need to fix this on 2.1.
